code/controllers/subsystem/non_firing/SSmapping.dm

```diff
@@ -118,6 +118,40 @@ SUBSYSTEM_DEF(mapping)
 	else
 		world.name = station_name()
 
+	// hehe
+	// try catch here so if it fails it doesn't actually kill the server
+	try
+		var/area/meeting_area
+		for(var/area/station/command/meeting_room/meet in world)
+			meeting_area = meet
+
+		if(!meeting_area)
+			// failing that
+			for(var/area/station/command/bridge/br in world)
+				meeting_area = br
+
+		if(!meeting_area)
+			return
+
+		var/atom/sur_le_table = locate(/obj/structure/table) in meeting_area
+		if(sur_le_table)
+			new /obj/structure/emergency_meeting_button(get_turf(sur_le_table))
+
+		if(prob(30))
+			var/list/station_areas = list()
+			for(var/area/station/ar in world)
```

Comment:
you should be able to pick from SSmapping.existing_station_areas above but unless there is a horrible performance beast it hardly matters

Reply:
Yeah since it happens only once I think it's fine, I tried pulling from globs but it didn't work